### PR TITLE
Add general support restore expansion

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: expansion/src/main/java/no/unit/nva/expansion/model/ExpandedGeneralSupportRequest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.Set;
+import no.unit.nva.expansion.ResourceExpansionService;
+import no.unit.nva.identifiers.SortableIdentifier;
+import no.unit.nva.publication.model.PublicationSummary;
+import no.unit.nva.publication.model.business.GeneralSupportRequest;
+import no.unit.nva.publication.service.impl.ResourceService;
+import nva.commons.apigateway.exceptions.NotFoundException;
+
+@JsonTypeName(ExpandedGeneralSupportRequest.TYPE)
+public class ExpandedGeneralSupportRequest implements ExpandedTicket {
```
Comment (axthosarouris) [inline expansion/src/main/java/no/unit/nva/expansion/model/ExpandedGeneralSupportRequest.java:16]: Currently the Expanded entry has insufficient data. After we have finished with the ticket service we should describe what we expect from an expanded ticket.

File: expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java [hunk 2]
```diff
@@ -209,36 +206,12 @@ void shouldAddResourceTitleToExpandedEntry(Class<?> type) throws ApiGatewayExcep
...
-    }
-    
-    private static boolean isTypeWithSubtypes(Type type) {
-        return type.value().getAnnotationsByType(JsonSubTypes.class).length > 0;
-    }
-    
-    private static List<Type> fetchDirectSubtypes(Class<?> type) {
-        var annotations = type.getAnnotationsByType(JsonSubTypes.class);
-        return Arrays.asList(annotations[0].value());
+    private static List<Class<?>> dataEntryTypeProvider() {
```
Comment (axthosarouris) [inline expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java:209]: replace duplicated code with the "library" code

File: expansion/src/test/java/no/unit/nva/expansion/model/ExpandedDataEntryTest.java [hunk 3]
```diff
@@ -192,11 +193,15 @@ public static ExpandedDataEntryWithAssociatedPublication create(
...
             } else if (expandedDataEntryClass.equals(ExpandedDoiRequest.class)) {
                 return new ExpandedDataEntryWithAssociatedPublication(publication, randomDoiRequest(publication,
-                    resourceExpansionService, messageService,ticketService));
+                    resourceExpansionService, messageService, ticketService));
             } else if (expandedDataEntryClass.equals(ExpandedPublishingRequest.class)) {
                 return new ExpandedDataEntryWithAssociatedPublication(publication,
                     createExpandedPublishingRequest(publication, resourceService, messageService,
                         resourceExpansionService));
+            } else if (expandedDataEntryClass.equals(ExpandedGeneralSupportRequest.class)) {
+                var result = new ExpandedDataEntryWithAssociatedPublication(publication,
```
Comment (brinxmat) [inline expansion/src/test/java/no/unit/nva/expansion/model/ExpandedDataEntryTest.java:202]: Return directly

**Review by brinxmat — APPROVED** [review]
See comment, a minor fluff

**Pull request merged**